flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/heap/AbstractHeapVector.java

```diff
@@ -18,14 +18,25 @@
 
 package org.apache.flink.table.dataformat.vector.heap;
 
-import org.apache.flink.table.dataformat.vector.AbstractColumnVector;
+import org.apache.flink.core.memory.MemoryUtils;
+import org.apache.flink.table.dataformat.vector.writable.AbstractWritableVector;
 
+import java.nio.ByteOrder;
 import java.util.Arrays;
 
 /**
  * Heap vector that nullable shared structure.
  */
-public abstract class AbstractHeapVector extends AbstractColumnVector {
+public abstract class AbstractHeapVector extends AbstractWritableVector {
+
+	public static final boolean LITTLE_ENDIAN = ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN;
+
+	public static final sun.misc.Unsafe UNSAFE = MemoryUtils.UNSAFE;
+	public static final int BYTE_ARRAY_OFFSET = UNSAFE.arrayBaseOffset(byte[].class);
```

Comment:
Just use `Unsafe::ARRAY_BYTE_BASE_OFFSET`?

Reply:
Use `arrayBaseOffset` is more safe for cross platform.